src/dispatch/participant/service.py

```diff
@@ -79,34 +79,43 @@ def get_all_by_incident_id(*, db_session, incident_id: int) -> List[Optional[Par
 def get_or_create(
     *,
     db_session,
-    incident_id: int,
+    subject_id: int,
+    subject_type: str,
     individual_id: int,
     service_id: int,
     participant_roles: List[ParticipantRoleCreate],
 ) -> Participant:
     """Gets an existing participant object or creates a new one."""
-    from dispatch.incident import service as incident_service
-
-    participant = (
+    participant: Participant = (
         db_session.query(Participant)
-        .filter(Participant.incident_id == incident_id)
+        .filter(
+            Participant.incident_id == subject_id
```

Comment:
Not a huge of fan of doing this inline due to readable concerns. Especially as the query building allows you to do this pretty easily. 

```
query = db_session.query(Participant)

if subject_type == 'incident':
     query = query.filter(Participant.incident_id == subject_id)
 else:
     query = query.filter(Participant.case_id == subject_id)
```